b2dEditor.py

```diff
@@ -0,0 +1,347 @@
+                        exec(self.command)
+                        button_flag = True
+
+        if event.type == pygame.MOUSEBUTTONUP:
+            if event.button == 1:
+                button_flag = False
+
+
+# Define Node class
+class Node():
+    def __init__(self):
+        self.norm_color = (0, 0, 200)
+        self.select_color = (0, 100, 0)
+        self.color = self.norm_color
+        self.pos = (0, 0)
+        self.rect = None
+
+    def set_pos(self, pos):
+        self.pos = pos
+
+    def set_color(self, color):
+        self.color = color
+
+class OriginNode():
+    def __init__(self):
+        self.color = (200, 0, 0)
+        self.pos = (0, 0)
+        self.rect = None
+
+    def set_pos(self, pos):
+        self.pos = pos
+
+    def set_color(self, color):
+        self.color = color
+
+class NodeRect():
+    def __init__(self):
+        self.edge_color = (150, 150, 150)
+        self.color = (250, 250, 250)
+        self.rect = pygame.Rect(50, 50, resolution[1] - 100, resolution[1] - 100)
+
+    def handle_mouse(self, event):
+        global selected
+
+        if event.type == pygame.MOUSEBUTTONDOWN:
+            if event.button == 1:
+                for nodes in shapes:
+                    for i in nodes:
+                        # Compare event position to each node's position
+                        if i.pos[0] - 5 <= event.pos[0] <= i.pos[0] + 5 and i.pos[1] - 5 <= event.pos[1] <= i.pos[1] + 5:
+                        # Set clicked node's color and set it as selected
+                            selected = i
+                            set_current_shape(shapes.index(nodes))
+
+                if origin.pos[0] - 6 <= event.pos[0] <= origin.pos[0] + 6 and origin.pos[1] - 6 <= event.pos[1] <= origin.pos[1] + 6:
+                    selected = origin
+
+                if selected == None:
+                    if edit_area.rect.collidepoint(mouse_pos):
+                        add_node(event.pos)
+
+            # Check if mouse2 (3 in pygame terms) clicked on node
+            elif event.button == 3:
+                for nodes in shapes:
+                    for i in nodes:
+                        # Compare event position to each node's position
+                        if i.pos[0] - 5 <= event.pos[0] <= i.pos[0] + 5 and i.pos[1] - 5 <= event.pos[1] <= i.pos[1] + 5:
+                        # Remove clicked node
+                            nodes.remove(i)
+
+        # Set selected node's position to the mouse's
+        elif event.type == pygame.MOUSEMOTION:
+            if selected != None:
+                if edit_area.rect.collidepoint(mouse_pos):
+                    selected.set_pos(event.pos)
+                else: selected = None
+
+
+        # De-select node on mouse1 release
+        elif event.type == pygame.MOUSEBUTTONUP:
+            if event.button == 1:
+                if selected != None:
+                    selected.clicked = False
+                selected = None
+
+
+def set_current_shape(num):
```

Comment:
Either have all functions before class declaration, or after. I'd suggest after, since for `run()`m it makes sense, and others should stay with it.